internal/parser.go

```diff
@@ -2553,9 +2556,6 @@ func inExpressionIM(p *parser) bool {
 		return true
 	}
 	p.im = p.originalIM
-	if p.im == nil {
-		p.im = inBodyIM
-	}
```

Comment:
Unrelated fix, but this could obscure a valid panic if author did not set `originalIM` before entering `inExpressionIM`.